internal/installation/receipt/receipt_test.go

```diff
@@ -73,6 +74,38 @@ func TestLoad_preservesNonExistsError(t *testing.T) {
 	}
 }
 
+func TestCanonicalName(t *testing.T) {
+	tests := []struct {
+		name     string
+		receipt  index.Receipt
+		expected string
+	}{
+		{
+			name:     "from default index",
+			receipt:  testutil.NewReceipt().WithPlugin(testutil.NewPlugin().WithName("foo").V()).V(),
+			expected: "foo",
+		},
+		{
```

Comment:
the missing test case here would be that there's an explicit status, it's from index `default`.